### DISCUSSION TITLE
Video in I/O section

### DISCUSSION THREAD

**Discussion by justingeller** [EnviralDesign/GeoPix#31] [Q&A]
I work exclusively with video as sources in the I/O tab but all the new tutorials almost exclusively deal with ramps and other generators as sources for the projectors. I created a macro and then used Create>Texture Node>Load>External File to add a video but can't seem to find a way to just get it to easily loop the exact length of the video. GeoPix 1.0 was pretty drag and drop with importing video and having it just automatically loop. Is this functionally not there yet or is there a tutorial coming that explains it easier than I can figure it out? Thanks!

**Comment by EnviralDesign** [comment]
Hey @justingeller !

Yeah there has been some extensive changes with the movie clips in v2,  to make them congruent with all the other generator style node that have a sense of timing. The thing that was missing really was some tools to quickly trim a clip or set a parent macro to the length of the clip, the little button I had was pretty outdated. 

I've made quite a few changes today, mostly surrounding timing tools with movie clips, and a few other bug fixes.
Give it a go and let me know!

The workflow goes something like this:
1. create a macro, go inside
2. drag a video clip into the macro's graph. 
3. the import procedure will auto trim the clip, if it's longer than the macro. you can turn the trimming off though if you want on the attributes, or customize the trim points. 
4. If for any reason you need to auto trim again, you can set a new Trim Start, and click "Trims from Macro". This button will ensure that the clips plays back 1:1 speed, making the clip loop more to compensate.

Trimming a clip essentially makes a subset of the clip, "the clip" effectively discarding anything outside of that time range. changing trims will also change the speed of the clip relative to how much is trimmed.

If you want to do the reverse, and set the Macro to the length of the full clip for instance and have 1:1 playback speed, do this:
1. toggle off "trim clip"
2. click "Macro from Trims" button.
3. Macro from Trims will use the length of the clip, and set the Macro to that length as well. If you have trimming enabled, it will use the trimmed length. if you have trimming disabled, it uses the native full length of the video instead.

I really need to make some tutorials on the IO nodes !
Hope this helps.

**Reply by justingeller** [reply]
Thanks! That explains it for me. It's a couple more hoops to jump through now on set up but it looks like this opens the door to way more flexibility in the future.

**Reply by EnviralDesign** [reply]
Ya one thing I think could be useful is a drag n drop callback where if you drag a video clip into the IO top level, it automatically creates a Macro, with a clip inside, and all configured accordingly, timing, length etc.

would this mesh with the workflow you use? I think it would be useful to many who just want to bring in a bunch of clips and start jamming.

**Reply by justingeller** [reply]
absolutely! Being able to drag and drop makes it super easy. Although, I do miss the clip bins that just auto-populate in the Perform tab like a crate of records to DJ. But I know that the Perform tab has a long way to go, so I'm not sweating it too much!

**Reply by EnviralDesign** [reply]
right on! good to know what's liked/missed from before. I have plans to bring something like that back, although I implement it a bit differently. but GeoPix does need a content bin that is a few clicks to get going with.
